File: Level-1/4. Arrays/java/Subarray_problem.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {
        // write your code here
        Scanner scn = new Scanner(System.in);
        int n = scn.nextInt();
        int arr[] = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = scn.nextInt();
        }
        subarray(arr);
    }
    public static void subarray(int arr[]) {
        int str = 0;
        while (str < arr.length) {
            int end = str;
            while (end < arr.length) {
                for (int i = str; i <= end; i++) {
                    System.out.print(arr[i] + "\t");
                }
                System.out.println();
                end++;
            }
            // System.out.println();
            str++;
        }
    }
}

File: Level-1/4. Arrays/cpp/Subarray_Problem.cpp
#include <iostream>
using namespace std;

void subarray(int *arr, int n)
{
    int str = 0;
    while (str < n)
    {
        int end = str;
        while (end < n)
        {
            for (int i = str; i <= end; i++)
            {
                cout << arr[i] << "\t";
            }
            cout << endl;
            end++;
        }
        str++;
    }
}

int main()
{
    int n;
    cin >> n;
    int *arr = new int[n];
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }

    subarray(arr, n);
}